scripts/tcl_commands/synthesis.tcl

```diff
@@ -113,6 +113,7 @@ proc run_synthesis {args} {
     puts_info "Running Synthesis (log: [relpath . $log])..."
 
     set ::env(CURRENT_SDC) $::env(BASE_SDC_FILE)
+    set ::env(SDC_IN) $::env(BASE_SDC_FILE)
```

Comment:
It appears that you're relying on the flow being run as a monolith for `SDC_IN` to have an appropriate value, i.e., if synthesis is not run then multiple PnR steps will fail. I'm wondering if it's appropriate to move the logic of picking between signoff/PnR SDC inside `read_current_sdc`